pkg/controller/siddhiprocess/deploymanager/application.go

```diff
@@ -0,0 +1,288 @@
+		if err != nil {
+			return
+		}
+		mountPath := d.Image.Home + DepConfMountPath
+		volume, volumeMount := createCMVolumes(deployYAMLCMName, mountPath)
+		volumes = append(volumes, volume)
+		volumeMounts = append(volumeMounts, volumeMount)
+		configParameter = DepConfParameter + mountPath + deployYAMLCMName
+	} else if d.SiddhiProcess.Spec.SiddhiConfig != "" {
+		deployYAMLCMName := d.SiddhiProcess.Name + DepCMExtension
+		data := map[string]string{
+			deployYAMLCMName: d.SiddhiProcess.Spec.SiddhiConfig,
+		}
+		err = d.KubeClient.CreateOrUpdateCM(deployYAMLCMName, d.SiddhiProcess.Namespace, data, d.SiddhiProcess)
+		if err != nil {
+			return
+		}
+		mountPath := d.Image.Home + DepConfMountPath
+		volume, volumeMount := createCMVolumes(deployYAMLCMName, mountPath)
+		volumes = append(volumes, volume)
+		volumeMounts = append(volumeMounts, volumeMount)
+		configParameter = DepConfParameter + mountPath + deployYAMLCMName
+	}
+
+	maxUnavailable := intstr.IntOrString{
+		Type:   artifact.Int,
+		IntVal: MaxUnavailable,
+	}
+	maxSurge := intstr.IntOrString{
+		Type:   artifact.Int,
+		IntVal: MaxSurge,
+	}
+	rollingUpdate := appsv1.RollingUpdateDeployment{
+		MaxUnavailable: &maxUnavailable,
+		MaxSurge:       &maxSurge,
+	}
+	depStrategy = appsv1.DeploymentStrategy{
+		Type:          appsv1.RollingUpdateDeploymentStrategyType,
+		RollingUpdate: &rollingUpdate,
+	}
+
+	if len(d.Application.Apps) > 0 {
+		appsCMName := d.Application.Name + strconv.Itoa(int(d.SiddhiProcess.Status.CurrentVersion))
+		for k, v := range d.Application.Apps {
+			key := k + SiddhiExtension
+			appsMap[key] = v
+		}
+		err = d.KubeClient.CreateOrUpdateCM(appsCMName, d.SiddhiProcess.Namespace, appsMap, d.SiddhiProcess)
+		if err != nil {
+			return
+		}
+		appsPath := d.Image.Home + SiddhiFilesDir
+		volume, volumeMount := createCMVolumes(appsCMName, appsPath)
+		volumes = append(volumes, volume)
+		volumeMounts = append(volumeMounts, volumeMount)
+		appParameter = AppConfParameter + appsPath + " "
+	} else {
+		appParameter = ParserParameter
+	}
+
+	userID := int64(802)
+	operationResult, _ = d.KubeClient.CreateOrUpdateDeployment(
+		d.Application.Name,
+		d.SiddhiProcess.Namespace,
+		d.Application.Replicas,
+		d.Labels,
+		d.Image.Name,
+		ContainerName,
+		[]string{Shell},
+		[]string{
+			d.Image.Home + SiddhiBin + "/" + d.Image.Profile + ".sh",
+			appParameter,
+			configParameter,
+		},
+		containerPorts,
+		volumeMounts,
+		d.SiddhiProcess.Spec.Container.Env,
+		corev1.SecurityContext{RunAsUser: &userID},
+		corev1.PullAlways,
+		imagePullSecrets,
+		volumes,
+		depStrategy,
+		d.SiddhiProcess,
+	)
+	return
+}
+
+// createLocalObjectReference creates a local object reference secret to download docker images from private registries.
+func createLocalObjectReference(secret string) (localObjectRef corev1.LocalObjectReference) {
+	localObjectRef = corev1.LocalObjectReference{
+		Name: secret,
+	}
+	return
```

Comment:
You can remove this return statement

Reply:
Fixed in https://github.com/siddhi-io/siddhi-operator/pull/123/commits/99c34b77761500db9edbf37569a84caa35738477